File: muxer-commons/src/main/java/pt/um/tf/taskmux/commons/task/AsyncTask.java
package pt.um.tf.taskmux.commons.task;

import pt.um.tf.taskmux.commons.URIGenerator;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutorService;

public abstract class AsyncTask<T> extends Task<CompletableFuture<Result<T>>> {
    private ExecutorService executorService;

    public AsyncTask(URIGenerator u) {
        super(u);
    }

    protected AsyncTask() {}

    public ExecutorService getExecutor() {
        return executorService;
    }

    public void setExecutor(ExecutorService executorService) {
        this.executorService = executorService;
    }
}

File: muxer-commons/src/main/kotlin/pt/um/tf/taskmux/commons/task/AsyncTask.kt
package pt.um.tf.taskmux.commons.task

import pt.um.tf.taskmux.commons.URIGenerator
import java.util.concurrent.CompletableFuture
import java.util.concurrent.ExecutorService
import java.util.concurrent.ForkJoinPool

abstract class AsyncTask<T>(u : URIGenerator = URIGenerator()) : Task<CompletableFuture<Result<T>>>(u) {
    var executor : ExecutorService? = null
}
